### DISCUSSION TITLE
`Invalid version. Must be a string. Got type "object"` since #23264

### DISCUSSION THREAD

**Discussion by astellingwerf** [renovatebot/renovate#26912] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

v37.153.2

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the problem

Between v37.141.0 and v37.153.2, most likely with #23264, scans started failing with error message `Invalid version. Must be a string. Got type "object"`.

See https://github.com/astellingwerf/renovate-reproduce-bug-from-23264 for a trivial reproduction.

### Relevant debug logs

<details><summary>Logs</summary>
See [job 697e3286-e1ee-4967-86ee-cddbb3b380d3](https://developer.mend.io/github/astellingwerf/renovate-reproduce-bug-from-23264/-/job/697e3286-e1ee-4967-86ee-cddbb3b380d3)

```

{
  "loggerErrors": [
    {
      "name": "renovate",
      "level": 50,
      "logContext": "QCwKHXgzok7QZxO-6gW1R",
      "repository": "astellingwerf/renovate-reproduce-bug-from-23264",
      "currentValue": ">=7.23.5",
      "datasource": "npm",
      "packageName": "babel-traverse",
      "followTag": null,
      "packageFile": "package.json",
      "pinDigests": false,
      "rollbackPrs": false,
      "updatePinnedDependencies": true,
      "unconstrainedValue": false,
      "err": {
        "message": "Invalid version. Must be a string. Got type \"object\".",
        "stack": "TypeError: Invalid version. Must be a string. Got type \"object\".\n    at new SemVer (/opt/containerbase/tools/renovate/37.153.2/node_modules/semver/classes/semver.js:19:13)\n    at compare (/opt/containerbase/tools/renovate/37.153.2/node_modules/semver/functions/compare.js:3:32)\n    at Object.eq [as equals] (/opt/containerbase/tools/renovate/37.153.2/node_modules/semver/functions/eq.js:2:29)\n    at /opt/containerbase/tools/renovate/37.153.2/node_modules/renovate/lib/workers/repository/process/lookup/index.ts:267:20\n    at Array.find (<anonymous>)\n    at lookupUpdates (/opt/containerbase/tools/renovate/37.153.2/node_modules/renovate/lib/workers/repository/process/lookup/index.ts:266:51)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at withLookupStats (/opt/containerbase/tools/renovate/37.153.2/node_modules/renovate/lib/workers/repository/process/fetch.ts:27:18)\n    at fetchDepUpdates (/opt/containerbase/tools/renovate/37.153.2/node_modules/renovate/lib/workers/repository/process/fetch.ts:72:30)\n    at /opt/containerbase/tools/renovate/37.153.2/node_modules/p-map/index.js:57:22"
      },
      "msg": "lookupUpdates error"
    }
  ]
}

```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction in the description above

**Comment by rarkins** [comment]
I run `pnpm start --dry-run=lookup --require-config=optional astellingwerf/renovate-reproduce-bug-from-23264 ` and don't see any error

**Reply by rarkins** [reply, marked as answer]
This might be the fix though: https://github.com/renovatebot/renovate/pull/26913

**Comment by rarkins** [comment]
We are wondering if there was a temporary bug in a public host like npmjs, because this problem came and went in our hosted app for a few hours only
